src/cmd/build.js

```diff
@@ -124,11 +139,29 @@ async function defaultPackageCreator({
   const packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   const extensionPath = path.join(artifactsDir, packageName);
-  const stream = createWriteStream(extensionPath);
+
+  //Added 'wx' flags to avoid overwriting of existing package
+  let stream = createWriteStream(extensionPath, {flags: 'wx'});
 
   stream.write(buffer, () => stream.end());
 
-  await eventToPromise(stream, 'close');
+  try {
+    await eventToPromise(stream, 'close');
+  } catch (error) {
+    if (isErrorWithCode('EEXIST', error)) {
```

Comment:
@harikishen By re-looking at this block I think that we can reduce the nesting of the `if` blocks by reversing the check and make it nicer:

```js
} catch (error) {
  if (!isErrorWithCode('EEXIST', error)) {
    throw error;
  }

  if (!overwriteDest) {
    ...
  }

  log.info(`Destination exists, overwriting: ${extensionPath}`);
  ...
} 
```